### PR TITLE
starter app now has a button that does something

### PR DESCRIPTION
**Pull request by sethladd**
Addresses https://github.com/flutter/flutter/issues/1152

### CODE REVIEW

**Comment by sethladd** [comment]
cc @devoncarew @abarth

File: packages/flutter_tools/lib/src/commands/init.dart [hunk 1]
```diff
@@ -201,14 +201,27 @@ void main() {
...
+      routes: <String, RouteBuilder>{
         '/': (RouteArguments args) => new FlutterDemo()
       }
     )
   );
 }
 
-class FlutterDemo extends StatelessComponent {
+class FlutterDemo extends StatefulComponent {
+  @override
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/init.dart]: do we want this annotation?

Comment (abarth) [inline packages/flutter_tools/lib/src/commands/init.dart]: We've been skipped it in the framework because there isn't great tooling support for it.

**Comment by abarth** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/init.dart (hunk 1 above)
Comment (sethladd) [inline packages/flutter_tools/lib/src/commands/init.dart]: Aside: curious what we want the tools to do.

If I use the annotation, I get a warning if my superclass doesn't contain the method

screen shot 2016-01-12 at 10 55 40 am

Which can help when code is refactored.

Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/init.dart]: To make it useful, we'd have to use it everywhere. To use it everywhere, we need the tools to complain when it's missing. They don't complain when it's missing.

Once we can do analyzer plugins that's one of the things I may add support for.

Comment (sethladd) [inline packages/flutter_tools/lib/src/commands/init.dart]: Ah, gotcha. We want a hint when `@override` isn't used. I'd also love to see a Quick Fix (add missing @override)

Comment (sethladd) [inline packages/flutter_tools/lib/src/commands/init.dart]: FWIW

https://github.com/dart-lang/sdk/issues/25416

and

https://github.com/dart-lang/sdk/issues/25417

**Pull request merged**